sdk/containerregistry/azure-containerregistry/azure/containerregistry/_exchange_client.py

```diff
@@ -0,0 +1,127 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import re
+from typing import TYPE_CHECKING
+
+from azure.core.pipeline.policies import SansIOHTTPPolicy
+
+from ._generated import ContainerRegistry
+from ._user_agent import USER_AGENT
+
+if TYPE_CHECKING:
+    from azure.core.credentials import TokenCredential
+    from azure.core.pipeline import PipelineRequest, PipelineResponse
+    from typing import Dict, Any, List
+
+
+class ExchangeClientAuthenticationPolicy(SansIOHTTPPolicy):
+    """Authentication policy for exchange client that does not modify the request"""
+
+    def on_request(self, request):
+        # type: (PipelineRequest) -> None
+        pass
+
+    def on_response(self, request, response):
+        # type: (PipelineRequest, PipelineResponse) -> None
+        pass
+
+
+class ACRExchangeClient(object):
+    """Class for handling oauth authentication requests
+
+    :param endpoint: Azure Container Registry endpoint
+    :type endpoint: str
+    :param credential: AAD Token for authenticating requests with Azure
+    :type credential: :class:`azure.identity.DefaultTokenCredential`
+
+    """
+
+    BEARER = "Bearer"
+    AUTHENTICATION_CHALLENGE_PARAMS_PATTERN = re.compile('(?:(\\w+)="([^""]*)")+')
+    WWW_AUTHENTICATE = "WWW-Authenticate"
+    SCOPE_PARAMETER = "scope"
+    SERVICE_PARAMETER = "service"
+    AUTHORIZATION = "Authorization"
```

Comment:
These are all part of the public API?
Or is this client not part of the public API?